src/libs/modulegroups.c

```diff
@@ -671,17 +717,1295 @@ static void _lib_modulegroups_switch_group(dt_lib_module_t *self, dt_iop_module_
+        list = dt_util_dstrcat(list, "|%s", op);
+      }
+      pos += op_len + 2;
+    }
+    list = dt_util_dstrcat(list, "|");
+    fav = dt_util_dstrcat(fav, "|");
+
+    gchar *tx = _preset_retrieve_old_layout(list, fav);
+    dt_lib_presets_add(pname, self->plugin_name, self->version(), tx, strlen(tx), FALSE);
+  }
+  sqlite3_finalize(stmt);
+}
+
+static gchar *_preset_to_string(GList *groups)
+{
+  gchar *res = NULL;
+  GList *l = groups;
+  res = dt_util_dstrcat(res, "ꬹ1");
+  while(l)
+  {
+    dt_lib_modulegroups_group_t *g = (dt_lib_modulegroups_group_t *)l->data;
+    res = dt_util_dstrcat(res, "ꬹ%s|%s|", g->name, g->icon);
+    GList *ll = g->modules;
+    while(ll)
+    {
+      gchar *m = (gchar *)ll->data;
+      res = dt_util_dstrcat(res, "|%s", m);
+      ll = g_list_next(ll);
+    }
+    l = g_list_next(l);
+  }
+  if(!res) res = g_strdup(" ");
+  return res;
+}
+
+static GList *_preset_from_string(gchar *txt)
+{
+  GList *res = NULL;
+  if(!txt) return res;
+
+  gchar **gr = g_strsplit(txt, "ꬹ", -1);
+  for(int i = 2; i < g_strv_length(gr); i++)
+  {
+    gchar *tx = gr[i];
+    if(tx)
+    {
+      gchar **gr2 = g_strsplit(tx, "|", -1);
+      const int nb = g_strv_length(gr2);
+      if(nb > 2)
+      {
+        dt_lib_modulegroups_group_t *group
+            = (dt_lib_modulegroups_group_t *)g_malloc0(sizeof(dt_lib_modulegroups_group_t));
+        group->name = g_strdup(gr2[0]);
+        group->icon = g_strdup(gr2[1]);
+        // gr2[2] is reserved for eventual future use
+        for(int j = 3; j < nb; j++)
+        {
+          group->modules = g_list_append(group->modules, g_strdup(gr2[j]));
+        }
+        res = g_list_append(res, group);
+      }
+      g_strfreev(gr2);
+    }
+  }
+  g_strfreev(gr);
+
+  return res;
+}
+
+void init_presets(dt_lib_module_t *self)
+{
+  gchar *tx = NULL;
+  tx = dt_util_dstrcat(
+      tx, "ꬹ1ꬹ%s|%s||%s", _("technical"), "technical",
+      "colorin|hazeremoval|filmicrgb|clipping|flip|lens|exposure|denoiseprofile|demosaic|highlights");
+  tx = dt_util_dstrcat(tx, "ꬹ%s|%s||%s", _("grading"), "grading", "rgblevels|colorbalance|toneequal|temperature");
+  tx = dt_util_dstrcat(tx, "ꬹ%s|%s||%s", _("effects"), "effect", "sharpen|bilat");
+  dt_lib_presets_add(_("default"), self->plugin_name, self->version(), tx, strlen(tx), TRUE);
+
+  gchar *tx2 = NULL;
+  tx2 = dt_util_dstrcat(tx2, "ꬹ1ꬹ%s|%s||%s", _("base"), "basic",
+                        "basecurve|toneequal|clipping|flip|exposure|demosaic|highlights|temperature|filmicrgb");
+  tx2 = dt_util_dstrcat(tx2, "ꬹ%s|%s||%s", _("tone"), "tone", "rgblevels|bilat");
+  tx2 = dt_util_dstrcat(tx2, "ꬹ%s|%s||%s", _("color"), "color", "colorbalance|colorin");
+  tx2 = dt_util_dstrcat(tx2, "ꬹ%s|%s||%s", _("correct"), "correct", "sharpen|hazeremoval|lens|denoiseprofile");
+  dt_lib_presets_add(_("legacy layout"), self->plugin_name, self->version(), tx2, strlen(tx2), TRUE);
+
+  // if needed, we add a new preset, based on last user config
```

Comment:
Ok, this is certainly why it is not working for me. modulegroups_preset is not currently used in master but was added when I tested this PR so now the import does not happens.

What I would propose is:
- always call the migration routine
- check for modulelist as done now with writeprotect=0
- when a user's modulelist is migrated I would remove it from data.db

To make this easier the call dt_conf_key_exists("plugins/darkroom/modulegroups_preset") could be replaced by a call checking if there is some user's modulelist in data.presets.

Reply:
Maybe even better, if we bump the module version we can probably do the migration of old presets into the legacy_params() code. This will be done once and follows the dt "way" of doing migration.

Reply:
Well, but not sure it is possible to use legacy_params() as the modulist module is removed.

Reply:
Indeed, as there's no pre-existing presets for modulegroups,  legacy_params is not called... That said, I've implemented your previous proposal and it works well !